index.js

```diff
@@ -14,6 +15,7 @@ const encodingNegotiator = require('@fastify/accept-negotiator')
 const dirList = require('./lib/dirList')
 
 async function fastifyStatic (fastify, opts) {
+  opts.root = normalizeRoot(opts.root)
```

Comment:
I would move this to `sendOptions` as we are editing the user's input

Reply:
What do you mean?

I would actually merge checkRootPathForErrors into normalizeRoot

Reply:
Checkrootforerrors is pretty much validation. Imo validation checks doesn't belong in a normalizing function that alters data.

Reply:
Please dont resolve discussions.

Reply:
Few lines down we can write `sendOptions = { root: normalizeRoot(opts.root)` without overwriting the user's input

<img width="450" alt="image" src="https://user-images.githubusercontent.com/11404065/224555107-91277db3-d267-4ee3-a589-94e5cc4864de.png">